node/src/components/storage.rs

```diff
@@ -1212,6 +1234,49 @@ fn check_block_metadata_db(
             ),
         }
     }
-    info!("Check for block_metadata_db complete");
+
+    drop(cursor);
+    txn.commit()?;
+
+    info!("block metadata database initialized");
+    Ok(())
+}
+
+/// Purges stale entries from the deploy metadata database.
+fn initialize_deploy_metadata_db(
+    env: &Environment,
+    deploy_metadata_db: &Database,
+    deleted_block_hashes: &HashSet<BlockHash>,
+) -> Result<(), LmdbExtError> {
```

Comment:
We should probably also check the integrity of the deploy data. Can you ticket that?

Reply:
I think we'd need to discuss that one first, just to agree the scope.  Do we just check the signature(s), or do we try and validate more than that?

Reply:
Sure, we can discuss the scope during the sprint planning but the blank ticket would serve as a reminder.

Reply:
Fair point - I'll make a ticket.